graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/rest/RuleResource.java

```diff
@@ -173,14 +190,51 @@ public RuleSource parse(@ApiParam(name = "rule", required = true) @NotNull RuleS
                 .build();
     }
 
+    @ApiOperation(value = "Simulate a single processing rule")
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Path("/simulate/{messageString}")
+    @NoAuditEvent("only used to test a rule, no changes made in the system")
+    public Message simulate(
+            @ApiParam(name = "rule", required = true) @NotNull RuleSource ruleSource,
+            @ApiParam(name = "messageString", required = true) @PathParam("messageString") @NotBlank String messageString
```

Comment:
We cannot pass the message string as a path parameter. That doesn't work for large values. And there might be escaping issues as well when passing in multi-line strings or structured data like JSON.

Please pass it in the request body. We can create a new request entity that contains the rule source and the message.